public/components/register_model/__tests__/model_task_manager.test.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { waitFor } from '../../../../test/test_utils';
+import { Task } from '../../../../public/apis/task';
+
+import { ModelTaskManager } from '../model_task_manager';
+
+describe('ModelTaskManager', () => {
+  const getOneMock = jest.fn();
+
+  beforeEach(() => {
+    jest.spyOn(Task.prototype, 'getOne').mockImplementation(getOneMock);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should call onComplete if model task complete successfully', async () => {
+    const onCompleteMock = jest.fn();
+    const onErrorMock = jest.fn();
+    const onUpdateMock = jest.fn();
+    const res = { model_id: 'model_id' };
+    getOneMock.mockResolvedValue(res);
+
+    const taskManager = new ModelTaskManager();
+    taskManager.query({
+      taskId: 'task_id',
+      onComplete: onCompleteMock,
+      onError: onErrorMock,
+      onUpdate: onUpdateMock,
+    });
+
+    await waitFor(() => {
+      expect(onCompleteMock).toHaveBeenCalledWith(res);
+      expect(onUpdateMock).toHaveBeenCalledWith(res);
+      expect(onErrorMock).not.toHaveBeenCalled();
+    });
+  });
+
+  it('should call onError if model task complete with error', async () => {
+    const onCompleteMock = jest.fn();
+    const onErrorMock = jest.fn();
+    const onUpdateMock = jest.fn();
+    const res = { error: 'error msg' };
+    getOneMock.mockResolvedValue(res);
+
+    const taskManager = new ModelTaskManager();
+    taskManager.query({
+      taskId: 'task_id',
+      onComplete: onCompleteMock,
+      onError: onErrorMock,
+      onUpdate: onUpdateMock,
+    });
+
+    await waitFor(() => {
+      expect(onCompleteMock).not.toHaveBeenCalled();
+      expect(onUpdateMock).not.toHaveBeenCalled();
+      expect(onErrorMock).toHaveBeenCalled();
+    });
```

Comment:
The `getOneMock` will be called twice, if we add test code like these:
```ts
    await new Promise((resolve) => {
      setTimeout(resolve, 2000);
    });
    expect(getOneMock).toHaveBeenCalledTimes(1); 
```
<img width="947" alt="image" src="https://user-images.githubusercontent.com/4034161/221087462-de49aee0-5c04-4b23-8a46-d4ed574b10f2.png">

Reply:
Nice catch, the polling should stop on `res.error`

Reply:
I think this model task manager may also used for model load, maybe we can move this model_task_manager outside in the future.

Reply:
Yes, the polling logic can be generic, we can refactor the code in the future.